src/task_manager.cc

```diff
@@ -232,7 +230,6 @@ namespace detail {
 		task* new_horizon;
 		{
 			auto reserve = m_task_buffer.reserve_task_entry(await_free_task_slot_callback());
-			std::lock_guard lock(m_task_mutex);
```

Comment:
Same thing regarding the extra compound statement + superfluous early definition applies here, and also in `generate_epoch_task`.

Reply:
Cleaned up.